src/Crdt.Core/QueryHelpers.cs

```diff
@@ -32,9 +32,11 @@ public static async Task<SyncState> GetSyncState(this IQueryable<CommitBase> com
                 var otherDt = DateTimeOffset.FromUnixTimeMilliseconds(otherTimestamp);
                 //todo even slower we want to also filter out changes that are already in the other history
```

Comment:
Isn't this what you just did? :)

Reply:
Yeah I guess that comment doesn't make sense. It's slow because there's 1 query per ClientId and that's what I'd like to be faster.